src/main/java/seedu/address/logic/commands/patient/SearchPatientCommand.java

```diff
@@ -56,12 +61,18 @@ public CommandResult execute(Model model) throws CommandException {
 
         final SearchCriteria criteriaToSearch = this.confirmCriteria(searchPatientDescriptor);
         List<Patient> patientList = model.getFilteredPatientList();
-
         if (criteriaToSearch == SearchCriteria.CRITERIA_NOT_FOUND) {
             throw new CommandException(MESSAGE_NOT_FOUND);
         } else if (criteriaToSearch == SearchCriteria.CRITERIA_IS_NAME) {
+            namePredicate = new NameContainsKeywordsPredicate(Arrays
+                    .asList(searchPatientDescriptor.getStringName().split("\\s+")));
+            model.updateFilteredPatientList(namePredicate);
             return findPatientWithName(searchPatientDescriptor, patientList);
         } else if (criteriaToSearch == SearchCriteria.CRITERIA_IS_TEMPERATURE) {
+            double startingTemperature = searchPatientDescriptor.getTemperatureRange().getStartingTemperature();
+            double endingTemperature = searchPatientDescriptor.getTemperatureRange().getEndingTemperature();
+            temperaturePredicate = new TemperatureRangePredicate(startingTemperature, endingTemperature);
+            model.updateFilteredPatientList(temperaturePredicate);
```

Comment:
Maybe, abstract away details, for better readability.

Reply:
Yup! Thanks for the advice!